lib/flake-utils.nix

```diff
@@ -0,0 +1,63 @@
+{ nixpkgs, systems }:
```

Comment:
(my fact are not as up to date as @Mic92 I think the pattern they use https://github.com/numtide/flake-utils/blob/master/simpleFlake.nix#L44 will be flawed in nixpkgs, because import <nixpkgs> kills some pipelines (I've been guilty of that :) ), as well as just being side-effecty patterns. I'd prefer to pass it as argument, but I'd be happy to hear other arguments on this.

Reply:
Coupling to nixpkgs in a broad sense is warranted. After all `stdenv` and its `mkDerivation` form the foundation of basically all Nix-lang based derivations out there, even if indirectly sometimes.
Enumerating the possible systems is definitely something `stdenv` should do and transposing an attrset does not constitute sufficient complexity to require a separate project.

I'm don't know if we should special-case `nixpkgs.legacyPackages` below though.

Reply:
@roberth I think you nicely resumed my concern in your other comments. Thx.

Reply:
flake-utils adds a lot of boilercode to work and needs to be included into almost every flake in the wild. Even the nix repo just invents a variants of it.